pkg/util/validation/limits.go

```diff
@@ -447,6 +448,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.Var(&l.EnabledPromQLExperimentalFunctions, "query-frontend.enabled-promql-experimental-functions", "Enable certain experimental PromQL functions, which are subject to being changed or removed at any time, on a per-tenant basis. Defaults to empty which means all experimental functions are disabled. Set to 'all' to enable all experimental functions.")
 	f.BoolVar(&l.Prom2RangeCompat, "query-frontend.prom2-range-compat", false, "Rewrite queries using the same range selector and resolution [X:X] which don't work in Prometheus 3.0 to a nearly identical form that works with Prometheus 3.0 semantics")
 	f.BoolVar(&l.SubquerySpinOffEnabled, "query-frontend.subquery-spin-off-enabled", false, "Enable spinning off subqueries from instant queries as range queries to optimize their performance.")
+	f.BoolVar(&l.LabelsQueryOptimizerEnabled, "query-frontend.labels-query-optimizer-enabled", false, "Enable labels query optimizations. When enabled, the query-frontend rewrites labels queries in an idempotent way to improve their performance.")
```

Comment:
does the idempotency matter to whoever is operating mimir? the way i read this is that the same optimization can be applied multiple times and will always lead to the same result.

Reply:
Done in https://github.com/grafana/mimir/pull/12054/commits/83776c2cb1cb986777feeab4ee52a3867281696f

Reply:
> In this case they want it off when they're pushing metrics without __name__ labels

As discussed offline, in the distributors we enforce that every metric has `__name__`. I think that currently there's no way to write metrics without name.

The reason why it's disabled by default is because I want to gain confidence about it, but my idea would be to switch the default to true once we feel comfortable.